### PR TITLE
fix(yarn): revert default proxy behavior

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Change Yarn proxy behavior to opt-in

## Context

Addresses #27983

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/readme.md [hunk 1]
```diff
@@ -17,9 +17,10 @@ If Renovate detects a `packageManager` setting for Yarn in `package.json` then i
 #### HTTP Proxy Support
 
 Yarn itself does not natively recognize/support the `HTTP_PROXY` and `HTTPS_PROXY` environment variables.
-If Renovate detects Yarn 2+, and one or both of those variables are present, then it will run commands like `yarn config set --home httpProxy http://proxy` prior to executing `yarn install`.
+
+You can configure `RENOVATE_X_YARN_PROXY=true` as an environment variable to enable configuring of Yarn proxy (e.g. if you already configure these proxy settings yourself in `~/.yarnrc.yml`);
```
Comment (viceice) [inline lib/modules/manager/npm/readme.md:21]: this sounds wrong

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/npm/readme.md (hunk 1 above)
Comment (rarkins) [inline lib/modules/manager/npm/readme.md:21]:
```suggestion
You can configure `RENOVATE_X_YARN_PROXY=true` as an environment variable to enable configuring of Yarn proxy (e.g. if you cannot configure these proxy settings yourself in `~/.yarnrc.yml`).
```

Comment (rarkins) [reply lib/modules/manager/npm/readme.md]: Fixed

**Review by viceice — APPROVED** [review]

**Pull request merged**